packages/botonic-nlu/src/preprocessor.ts

```diff
@@ -0,0 +1,119 @@
+/* eslint-disable @typescript-eslint/unbound-method */
+import { DefaultTokenizer } from './preprocessing-tools/tokenizer';
+import { DefaultStemmer } from './preprocessing-tools/stemmer';
+import { DefaultNormalizer } from './preprocessing-tools/normalizer';
+import { UNKNOWN_TOKEN } from './constants';
+import { DataSet, Vocabulary, Normalizer, Stemmer, Tokenizer } from './types';
+import { Language } from './language';
+
+export class Preprocessor {
+  private _normalizer: Normalizer;
+  private _stemmer: Stemmer;
+  private _tokenizer: Tokenizer;
+  private _language: Language;
+  private _maxSeqLen: number;
+  private _vocabulary: Vocabulary;
+
+  constructor() {
+    this._normalizer = new DefaultNormalizer();
+    this._tokenizer = new DefaultTokenizer();
+    this._stemmer = new DefaultStemmer();
+  }
+
+  set normalizer(value: Normalizer) {
```

Comment:
set is normally used to do extra work when a field is set. If you're doing anything else, I think it's less verbose to just declare the field as public